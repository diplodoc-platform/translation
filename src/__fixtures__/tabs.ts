const markdown = `\
{% list tabs %}

- Первый Таб

  Текст первого таба

  - список внутри таба
        
    контент внутри списка
- Второй Таб
  
  Текст второго таба

{% endlist %}

{% list tabs %}

* Первый Таб во второй таб секции

  контент первого таба во второй таб секции

{% endlist %}
`;

const skeleton = `\
{% list tabs %}

- %%%1%%%

  %%%2%%%

  - %%%3%%%
        
    %%%4%%%
- %%%5%%%
  
  %%%6%%%

{% endlist %}

{% list tabs %}

* %%%7%%%

  %%%8%%%

{% endlist %}
`;

const translations = new Map<string, string>([
    ['1', 'First tab'],
    ['2', 'text of the first tab'],
    ['3', 'list inside the tab'],
    ['4', 'content inside the list'],
    ['5', 'Second tab'],
    ['6', 'text of the second tab'],
    ['7', 'First tab inside the second tab section'],
    ['8', 'content of the first tab inside the second tab section'],
]);

export {markdown, skeleton, translations};
export default {markdown, skeleton, translations};
