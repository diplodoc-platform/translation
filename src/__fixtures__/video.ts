const markdown = `\
@[youtube](https://www.youtube.com/video?v=dQw4w9WgXcQ) текст после
@[youtube](dQw4w9WgXcQ)
`;

const skeleton = `\
@[youtube](https://www.youtube.com/video?v=dQw4w9WgXcQ) %%%1%%%
@[youtube](dQw4w9WgXcQ)
`;

const translations = new Map<string, string>([['1', 'text after']]);

export {markdown, skeleton, translations};
export default {markdown, skeleton, translations};
