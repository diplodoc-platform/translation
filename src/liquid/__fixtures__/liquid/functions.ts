import {Fixtures} from './fixture';

const functions: Fixtures = [
    {
        section: 'functions',
        content: 'Hello P{{ user.name.slice(1, 2) }}vel!',
        number: 1,
    },
    {
        section: 'functions',
        content: 'Sentence {{user.name.slice(1)}} content after.',
        number: 2,
    },
    {
        section: 'functions',
        content: '{{user.slice(1)}}.',
        number: 3,
    },
    {
        section: 'functions',
        content: '{{user_name-cool.slice(1)}}.',
        number: 4,
    },
];

export {functions};
export default {functions};
