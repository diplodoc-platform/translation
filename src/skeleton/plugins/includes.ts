import type MdIt from 'markdown-it';
import type State from 'markdown-it/lib/rules_core/state_core';
import type StateInline from 'markdown-it/lib/rules_inline/state_inline';
import {token} from 'src/utils';

const INCLUDE_REGEXP = /^{%\s*include\s*(notitle)?\s*\[(.+?)]\((.+?)\)\s*%}$/;
const INCLUDE_REGEXP_2 = /^{%\s*include\s*(notitle)?\s*\[(.+?)]\((.+?)\)\s*%}/;

export default function (md: MdIt) {
  const block = (state: State) => {
    const {tokens} = state;
    let i = 0;

    while (i < tokens.length) {
      const openToken = tokens[i];
      const contentToken = tokens[i + 1];
      const closeToken = tokens[i + 2];

      if (
        openToken.type === 'paragraph_open' &&
        contentToken.type === 'inline' &&
        contentToken.content.match(INCLUDE_REGEXP) &&
        closeToken.type === 'paragraph_close'
      ) {
        contentToken.children = [
          token('liquid', {
            content: '',
            skip: contentToken.content,
            markup: contentToken.content,
            subtype: 'Include',
          }),
        ];

        i += 3;
      } else {
        i++;
      }
    }
  };

  const inline = (state: StateInline) => {
    const ch1 = state.src[state.pos];
    const ch2 = state.src[state.pos + 1];

    if (ch1 + ch2 === '{%') {
      const match = INCLUDE_REGEXP_2.exec(state.src.slice(state.pos));

      if (match) {
        state.pos += match[0].length;

        const token = state.push('liquid', '', 0) as Token;
        token.skip = match[0];
        token.markup = match[0];
        token.subtype = 'Include';

        return true;
      }
    }

    return false;
  };

  md.core.ruler.before('linkify', 'includes', block);
  md.inline.ruler.before('linkify', 'includes', inline);
}
