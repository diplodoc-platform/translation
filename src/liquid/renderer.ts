import {Token, TokenType} from 'src/liquid';

import {generateX} from 'src/xlf/generator';

export type RendererRuleSet = Record<TokenType | string, RendererRule>;

export type RendererRule = (token: Token) => string;

class Renderer {
    private rendered: string;
    private tokens: Token[];
    private rules: RendererRuleSet;

    constructor(tokens: Token[]) {
        this.tokens = tokens;
        this.rendered = '';

        this.rules = {
            Text: this.plainText.bind(this),
            Space: this.plainText.bind(this),
            If: this.wrapInX.bind(this),
            Else: this.wrapInX.bind(this),
            EndIf: this.wrapInX.bind(this),
            ForInLoop: this.wrapInX.bind(this),
            EndForInLoop: this.wrapInX.bind(this),
            Function: this.wrapInX.bind(this),
            Filter: this.wrapInX.bind(this),
        };
    }

    render() {
        for (const token of this.tokens) {
            const rule = this.rules[token.type];
            if (!rule) {
                throw new Error(`unexpected token: ${token.type}`);
            }

            this.rendered += rule(token);
        }

        return this.rendered;
    }

    private plainText(token: Token) {
        return token.value;
    }

    private wrapInX(token: Token) {
        const {type, value} = token;

        return generateX({
            ctype: `liquid_${type}`,
            equivText: value,
        });
    }
}

export {Renderer};
export default {Renderer};
