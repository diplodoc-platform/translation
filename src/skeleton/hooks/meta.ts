import type {CustomRenderer} from 'src/renderer';
import type {Consumer} from 'src/consumer';
import {Liquid} from '../liquid';
import {token} from 'src/utils';
import MarkdownIt from 'markdown-it';

type MarkdownItWithMeta = MarkdownIt & {
  meta: {};
};

export function meta(this: CustomRenderer<Consumer, MarkdownItWithMeta>) {
  const meta = this.md.meta ?? {};

  if (!Object.keys(meta).length) {
    return '';
  }

  traverse(meta, (value, key) => {
    const tokenizer = new Liquid(value);
    this.state.process(token('fake', {skip: key}));
    this.state.process([...tokenizer.tokenize()]);
  });

  return '';
}

type Meta =
  | string
  | {
      [prop: string | number]: Meta;
    };

function traverse(
  meta: Meta,
  fn: (val: string, key: string | null | undefined) => void,
  key?: string | null,
) {
  if (typeof meta === 'string') {
    fn(meta, key);
  } else if (Array.isArray(meta)) {
    for (const val of meta) {
      traverse(val, fn);
    }
  } else if (meta && typeof meta === 'object') {
    for (const [key, val] of Object.entries(meta)) {
      traverse(val, fn, key);
    }
  }
}
