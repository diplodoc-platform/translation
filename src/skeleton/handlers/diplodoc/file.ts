import {MarkdownRenderer} from '@diplodoc/markdown-it-markdown-renderer';
import Token from 'markdown-it/lib/token';

import {replacer} from 'src/skeleton/replacer';

import {SkeletonRendererState} from 'src/skeleton/renderer';

function yfmFile(this: MarkdownRenderer<SkeletonRendererState>, tokens: Token[], i: number) {
    const token = tokens[i];
    const content = token.content;

    if (!content?.length) {
        return '';
    }

    token.attrSet('download', replacer(content, this.state));

    return '';
}

export {yfmFile};
export default {yfmFile};
