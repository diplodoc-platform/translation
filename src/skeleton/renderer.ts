import MarkdownIt from 'markdown-it';
import {
    mdRenderer,
    MarkdownRenderer,
    MarkdownRendererParams,
    MarkdownRendererEnv,
} from '@diplodoc/markdown-it-markdown-renderer';

// configure with diplodoc plugins
// @ts-ignore
import meta from 'markdown-it-meta';
// @ts-ignore
import sup from 'markdown-it-sup';
import notes from '@doc-tools/transform/lib/plugins/notes';
import cut from '@doc-tools/transform/lib/plugins/cut';
import checkbox from '@doc-tools/transform/lib/plugins/checkbox';
import anchors from '@doc-tools/transform/lib/plugins/anchors';
import monospace from '@doc-tools/transform/lib/plugins/monospace';
import imsize from '@doc-tools/transform/lib/plugins/imsize';
import file from '@doc-tools/transform/lib/plugins/file';
import includes from '@doc-tools/transform/lib/plugins/includes';
import tabs from '@doc-tools/transform/lib/plugins/tabs';
import video from '@doc-tools/transform/lib/plugins/video';
import table from '@doc-tools/transform/lib/plugins/table';

import skeletonHandlers from './handlers';
import hooks, {HooksParameters, HooksState} from './hooks';
import {rules} from './rules';

export type SkeletonRendererState = HooksState;

export type RenderParameters = BaseParameters & DiplodocParameters;
export type BaseParameters = {
    markdown: string;
};
export type DiplodocParameters = {
    lang?: string;
};

function render(parameters: RenderParameters) {
    if (!validParameters(parameters)) {
        throw new Error('invalid parameters');
    }

    const {markdown, lang} = parameters;

    const md = new MarkdownIt({html: true}) as HooksParameters['markdownit'];
    const env: MarkdownRendererEnv = {source: markdown.split('\n')};

    const {handlers} = skeletonHandlers.generate();

    const skeletonHooks = hooks.generate({markdownit: md});

    // todo: add non-destructive way of extending markdown-renderer hooks
    for (const defaultHook of Object.entries(MarkdownRenderer.defaultHooks)) {
        const [lifecycle, hooks_] = defaultHook as any;

        if (skeletonHooks.hooks[lifecycle]) {
            skeletonHooks.hooks[lifecycle] = [...skeletonHooks.hooks[lifecycle], ...hooks_];
        }
    }

    const mdOptions: MarkdownRendererParams<HooksState> = {
        handlers,
        initState: skeletonHooks.initState,
        rules,
        hooks: skeletonHooks.hooks,
    };
    const diplodocOptions = {
        lang: lang ?? 'ru',
        path: '',
    };

    // diplodoc plugins
    md.use(meta, diplodocOptions);
    md.use(notes, diplodocOptions);
    md.use(cut, diplodocOptions);
    md.use(sup, diplodocOptions);
    md.use(checkbox, diplodocOptions);
    md.use(anchors, diplodocOptions);
    md.use(monospace, diplodocOptions);
    md.use(imsize, diplodocOptions);
    md.use(file, diplodocOptions);
    md.use(includes, diplodocOptions);
    md.use(tabs, diplodocOptions);
    md.use(video, diplodocOptions);
    md.use(table, diplodocOptions);

    md.use(mdRenderer, mdOptions);

    return md.render(markdown, env);
}

function validParameters(parameters: RenderParameters) {
    const {markdown, lang} = parameters;

    const markdownCondition = markdown !== undefined;
    const langCondition = lang === undefined || lang === 'ru' || lang === 'en';
    const conditions = [markdownCondition, langCondition];

    return conditions.reduce((a, v) => a && v, true);
}

export {render, validParameters};
export default {render, validParameters};
