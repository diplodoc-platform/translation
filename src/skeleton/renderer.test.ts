import {render, RenderParameters} from './renderer';

import basic from 'src/__fixtures__/basic';
import variables from 'src/__fixtures__/variables';
import filters from 'src/__fixtures__/filters';
import functions from 'src/__fixtures__/functions';
import conditionals from 'src/__fixtures__/conditionals';
import loops from 'src/__fixtures__/loops';
import meta from 'src/__fixtures__/meta';
import notes from 'src/__fixtures__/notes';
import cuts from 'src/__fixtures__/cuts';
import gfmTables from 'src/__fixtures__/gfm-tables';
import sup from 'src/__fixtures__/sup';
import checkbox from 'src/__fixtures__/checkbox';
import anchors from 'src/__fixtures__/anchors';
import monospace from 'src/__fixtures__/monospace';
import imsize from 'src/__fixtures__/imsize';
import file from 'src/__fixtures__/file';
import links from 'src/__fixtures__/links';
import includes from 'src/__fixtures__/includes';
import strikethrough from 'src/__fixtures__/strikethrough';
import tabs from 'src/__fixtures__/tabs';
import video from 'src/__fixtures__/video';
import multilineTables from 'src/__fixtures__/multiline-tables';

describe('smoke', () => {
    test('it works', () => {
        const parameters = {
            markdown: '1',
        };

        render(parameters);
    });
});

describe('validates parameters', () => {
    it('works with valid parameters', () => {
        const parameters = {
            markdown: '1',
        };

        render(parameters);
    });

    it('throws on invalid parameters', () => {
        const parameters = {} as RenderParameters;

        expect(() => render(parameters)).toThrow();
    });
});

describe('skeleton rendering', () => {
    it('renders hashes instead of content of the text tokens', () => {
        const parameters = {
            markdown: basic.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with liquid variables', () => {
        const parameters = {
            markdown: variables.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with liquid filters', () => {
        const parameters = {
            markdown: filters.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with liquid functions', () => {
        const parameters = {
            markdown: functions.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with liquid conditionals', () => {
        const parameters = {
            markdown: conditionals.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with loops', () => {
        const parameters = {
            markdown: loops.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with meta', () => {
        const parameters = {
            markdown: meta.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with notes', () => {
        const parameters = {
            markdown: notes.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with cuts', () => {
        const parameters = {
            markdown: cuts.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with gfm tables', () => {
        const parameters = {
            markdown: gfmTables.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with sup text', () => {
        const parameters = {
            markdown: sup.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with checkboxes in text', () => {
        const parameters = {
            markdown: checkbox.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with anchors in text', () => {
        const parameters = {
            markdown: anchors.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with monospace text', () => {
        const parameters = {
            markdown: monospace.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with imsize images', () => {
        const parameters = {
            markdown: imsize.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with file in text', () => {
        const parameters = {
            markdown: file.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('does not render hash instead of the special syntax {#T} inside link text', () => {
        const parameters = {
            markdown: links.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('does not render hashes instead of the {% include ... %} syntax', () => {
        const parameters = {
            markdown: includes.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with strikethrough', () => {
        const parameters = {
            markdown: strikethrough.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with tabs', () => {
        const parameters = {
            markdown: tabs.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with video', () => {
        const parameters = {
            markdown: video.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders hash instead of the content from markdown with multiline tables', () => {
        const parameters = {
            markdown: multilineTables.markdown,
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });
});
