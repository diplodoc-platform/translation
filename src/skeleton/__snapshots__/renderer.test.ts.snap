// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`skeleton rendering does not render hash instead of the special syntax {#T} inside link text 1`] = `"%%%1%%% [{#T}](file.md \\"%%%2%%%\\") %%%3%%%"`;

exports[`skeleton rendering does not render hashes instead of the {% include ... %} syntax 1`] = `
"{% include notitle [Description](../_includes/file.md) %}

{% include [Description](../_includes/file.md) %}"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with anchors in text 1`] = `
"# %%%1%%%
%%%2%%%
## %%%3%%% {#anchor}"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with checkboxes in text 1`] = `
"[ ] %%%1%%%

[x] %%%2%%%
- [_] %%%3%%%
- [-] %%%4%%%
- [ ] %%%5%%%
- [x] %%%6%%%
- [X] %%%7%%%

- %%%8%%%
  - [x] %%%9%%%
  - [ ] %%%10%%%"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with cuts 1`] = `
"
{% cut \\"%%%1%%%\\" %}

%%%2%%%

{% endcut %}"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with file in text 1`] = `"{% file src=\\"sample-file\\" name=\\"%%%1%%%\\" type=\\"application/zip\\" %} %%%2%%% {% file src=\\"sample-file\\" name=\\"%%%3%%%\\" type=\\"application/zip\\" %}"`;

exports[`skeleton rendering renders hash instead of the content from markdown with gfm tables 1`] = `
"|%%%1%%%|%%%2%%%|
|-----------|-----------|
|[%%%3%%%](https://www.google.com)|**%%%4%%%**|
|![%%%5%%%](image.png \\"%%%6%%%\\")|*%%%7%%%*|
"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with imsize images 1`] = `"%%%1%%% ![%%%2%%%](image.png \\"%%%3%%%\\" =13x37) %%%4%%%"`;

exports[`skeleton rendering renders hash instead of the content from markdown with liquid conditionals 1`] = `
"# %%%1%%%
{% if OS == 'iOS' %}

%%%2%%% [%%%3%%%](https://www.apple.com/ios/app-store/)%%%4%%%

{% else %}

%%%5%%% [%%%6%%%](https://play.google.com)%%%7%%%

%%%8%%% {% if  OS == 'iOS' %} %%%9%%% {% else %} %%%10%%% {% endif %} %%%11%%%

{% endif %}"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with liquid filters 1`] = `
"# %%%1%%%
%%%2%%%

%%%3%%%"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with liquid functions 1`] = `
"%%%1%%%

%%%2%%%

%%%3%%%"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with liquid variables 1`] = `
"# %%%1%%%
%%%2%%%

%%%3%%% [%%%4%%%](files/file.md \\"%%%5%%%\\")

![%%%6%%%](files/image.png \\"%%%7%%%\\")"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with loops 1`] = `
"# %%%1%%%
{% for element in array %}

%%%2%%%

{% endfor %}

%%%3%%% {% for element in array %} %%%4%%% {% endfor %} %%%5%%%"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with meta 1`] = `
"---
title: '%%%1%%%'
description: '%%%2%%% %%%3%%% %%%4%%%'
key: |
  %%%5%%%
seq:
  - '%%%6%%%'
  - '%%%7%%%'
  - '%%%8%%%'
  - '%%%9%%%'
  - '%%%10%%%'
  - '%%%11%%%'
seq_of_mappings:
  - name: '%%%12%%%'
    age: 13
  - name: '%%%13%%%'
    age: 99
object_inline:
  name: '%%%14%%%'
  age: 33
array_inline:
  - '%%%15%%%'
  - '%%%16%%%'
  - '%%%17%%%'
  - '%%%18%%%'
seq_of_seq:
  - - '%%%19%%%'
    - 65
    - 0.278
  - - '%%%20%%%'
    - 63
    - 0.288
nesting:
  - children:
      - name: '%%%21%%%'
      - name: '%%%22%%%'
---
# %%%23%%%
%%%24%%% %%%25%%%"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with monospace text 1`] = `
"##%%%1%%%##
%%%2%%% ##%%%3%%%## %%%4%%%**%%%5%%%** %%%6%%%##%%%7%%%##%%%8%%%
[%%%9%%% ##%%%10%%%## %%%11%%% ##%%%12%%%##](href \\"%%%13%%%\\")"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with multiline tables 1`] = `
"#|
||
**%%%1%%%**
|
**%%%2%%%**
||
||
%%%3%%%
|
%%%4%%%
||
|#

- %%%5%%%

  #|
  ||
  %%%6%%%
  |
  %%%7%%%
  ||
  ||
  %%%8%%%
  |
  %%%9%%%
  ||
  |#

  %%%10%%%"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with notes 1`] = `
"
{% note info \\"%%%1%%%\\" %}

%%%2%%%

{% endnote %}
- %%%3%%%

  {% note alert \\"%%%4%%%\\" %}

  %%%5%%%

  {% endnote %}"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with strikethrough 1`] = `
"~~%%%1%%%~~
[%%%2%%% ~~%%%3%%%~~ %%%4%%% ~~%%%5%%%~~](href \\"%%%6%%%\\")"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with sup text 1`] = `
"%%%1%%%^%%%2%%%^ %%%3%%%
%%%4%%%^%%%5%%%^ %%%6%%%
[%%%7%%%^%%%8%%%^ %%%9%%%](file.md)"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with tabs 1`] = `
"{% list tabs %}

- %%%1%%%

  %%%2%%%
  - %%%3%%%
    
    %%%4%%%
- %%%5%%%

  %%%6%%%

{% endlist %}


{% list tabs %}

* %%%7%%%

  %%%8%%%

{% endlist %}
"
`;

exports[`skeleton rendering renders hash instead of the content from markdown with video 1`] = `
"@[youtube](https://www.youtube.com/video?v=dQw4w9WgXcQ) %%%1%%%
@[youtube](dQw4w9WgXcQ)"
`;

exports[`skeleton rendering renders hashes instead of content of the text tokens 1`] = `
"# %%%1%%%
%%%2%%%
- %%%3%%%
- %%%4%%%
%%%5%%%
-----------
%%%6%%% **%%%7%%%** %%%8%%% *%%%9%%%*

%%%10%%% %%%11%%%
> %%%12%%%
1. %%%13%%%
2. %%%14%%%

[%%%15%%% %%%16%%%](files/file.md \\"%%%17%%% %%%18%%%\\")

![%%%19%%% %%%20%%%](files/image.png \\"%%%21%%% %%%22%%%\\")"
`;
